### PR TITLE
[ADP-3412] Cleanup old artifacts

### PR DESCRIPTION
**Pull request by paolino**
ADP-3412

### CODE REVIEW

File: lib/buildkite/exe/artifacts-csv.hs [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+organizationSlug :: Text
+organizationSlug = "cardano-foundation"
+
+mainPipeline :: PipelineName
+mainPipeline = PipelineName "cardano-wallet"
+
+adminPipeline :: PipelineName
+adminPipeline = PipelineName "cardano-wallet-buildkite-admin"
+
+buildkiteDomain :: String
```
Comment (paweljakubas) [inline lib/buildkite/exe/artifacts-csv.hs]: why `String` and not `Text` here?

File: lib/buildkite/exe/artifacts-csv.hs [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+buildkiteDomain = "api.buildkite.com"
+
+buildkitePort :: Int
+buildkitePort = 443
+
+newtype PipelineName = PipelineName Text
```
Comment (paweljakubas) [inline lib/buildkite/exe/artifacts-csv.hs]: why `String` and not `Text` here?

File: lib/buildkite/exe/artifacts-csv.hs [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+        runQuery f action = f $ runClientM action env
+        queryCardanoWallet = Query (runQuery bailout)
+            (withAuthWallet mainPipeline lock bkToken)
+        queryAdmin = Query (runQuery bailout)
+            (withAuthWallet adminPipeline lock bkToken)
+    _checkPoint <- S.head_
+        $ getBuilds queryAdmin
+        & S.chain (traceWith logsTracer . ProgressAdmin . number)
+        & flip S.for (getArtifacts queryAdmin)
+        & S.filter ((==) "finished" . state . snd)
```
Comment (paweljakubas) [inline lib/buildkite/exe/artifacts-csv.hs]: are two  `S.filter` calls result in two "runs" or it is "merged into one call"?  Probably not a problem here but would be good to be aware how it behaves. btw. we have `beautiful folds approach` that do many things at once in ONE run - see for example https://hackage.haskell.org/package/folds and https://www.youtube.com/watch?v=6a5Ti0r8Q2s

File: lib/buildkite/exe/artifacts-csv.hs [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+        & S.filter ((==) "artifacts-history/artifacts.csv" . path . snd)
+        & S.map snd
+    -- until
+    rows <-
+        S.foldMap_ (encodeNamedRecord artifactsHeader)
+            $ getBuilds queryCardanoWallet
+            & S.takeWhile (\b -> maybe True (number b >) mUntilBuild)
+            & S.chain (traceWith logsTracer . ProgressMain . number)
+            & flip S.for (getArtifacts queryCardanoWallet)
+            & S.filter ((==) "finished" . state . snd)
```
Comment (paweljakubas) [inline lib/buildkite/exe/artifacts-csv.hs:136]: the same here filer and map at once should be possible ...

File: lib/buildkite/src/Buildkite/API.hs [hunk 5]
```diff
@@ -22,30 +24,39 @@ module Buildkite.API
...
-    ( Parser
+    , ToJSON (..)
     )
 import Data.Data
     ( Proxy (..)
     )
 import Data.Text
     ( Text
     )
+import qualified Data.Text as T
```
Comment (paweljakubas) [inline lib/buildkite/src/Buildkite/API.hs:55]: wrong place - qualified imports are expected to be in separate section below.

File: lib/buildkite/src/Buildkite/API.hs [hunk 6]
```diff
@@ -69,26 +83,31 @@ newtype Time = Time
...
     { id :: Text
     , name :: Maybe Text
     , step_key :: Maybe Text
     , finished_at :: Maybe Time
+    , created_at :: Maybe Time
     }
     deriving (Show, Generic)
 
 instance FromJSON Job
+instance ToJSON Job
```
Comment (paweljakubas) [inline lib/buildkite/src/Buildkite/API.hs:107]: cannot use `via`  here?

File: lib/buildkite/src/Buildkite/API.hs [hunk 7]
```diff
@@ -172,22 +197,48 @@ type GetArtifact mime content =
...
-            :> Get '[mime] content
+            :> Get '[mime] (BuildKiteHeaders content)
         )
 
-type WithAuthPipeline a = Maybe Text -> Text -> Text -> a
+type WithAuthPipeline a = LimitsLock -> Maybe Text -> Text -> Text -> a
+
+respectLimits :: LimitsLock -> BuildKiteHeaders o -> ClientM o
+respectLimits l r = do
+    let remaining = case lookupResponseHeader r
```
Comment (paweljakubas) [inline lib/buildkite/src/Buildkite/API.hs:207]: hmmm, here we probably be better ;-)

File: lib/buildkite/src/Buildkite/API.hs [hunk 8]
```diff
@@ -172,22 +197,48 @@ type GetArtifact mime content =
...
+                        :: ResponseHeader "RateLimit-Reset" Int of
+            Header reset' -> reset'
+            _ -> error "RateLimit-Reset header not found"
+    liftIO $ setLimit l remaining reset
+    pure $ getResponse r
+
+withLimitsLock :: LimitsLock -> ClientM (BuildKiteHeaders b) -> ClientM b
+withLimitsLock l f = do
+    liftIO $ checkLimit l
+    r <- f
```
Comment (paweljakubas) [inline lib/buildkite/src/Buildkite/API.hs:221]: why not `f >>=respectLimits l` ?

File: lib/buildkite/src/Buildkite/Artifacts.hs [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+    , row_path :: Text
+    , row_created_at :: Time
+    , row_build :: Int
+    , row_job_id :: Text
+    }
+    deriving (Show)
+
+instance ToNamedRecord ArtifactRow where
+    toNamedRecord
+        ArtifactRow
```
Comment (paweljakubas) [inline lib/buildkite/src/Buildkite/Artifacts.hs:62]: what about using `RecordWildCards` here?

File: lib/benchmarks/exe/benchmark-history.hs [hunk 10]
```diff
@@ -245,7 +262,8 @@ main = do
     let env = buildkiteEnv manager
         runQuery :: HandleClientError a -> ClientM a -> IO (Maybe a)
         runQuery f action = f $ runClientM action env
-    result <- queryBuildkite runQuery (withAuthWallet bkToken) sinceDay
+    limitsLock <- newLimitsLock (show >$< stdoutTracer) 10
```
Comment (paweljakubas) [inline lib/benchmarks/exe/benchmark-history.hs]: :gem:

**Comment by paolino** [comment]
closed as the buildkite objective has changed (no need for cleanup artifacts) . Most of the code is reused in #4755

**Pull request closed without merging**